src/components/forms/Form.js

```diff
@@ -278,33 +298,80 @@ class Form extends Component {
 				<Button
 					title = "Cancel"
 					onPress = { () => {
+						this.resetState();
 						changeVisibility(false);
-						if (onCancel) onCancel();
 					} }
 				/>
 			</ButtonLayout>
 		);
 	}
 
-	submit() {
+	submit = () => {
 		let formIsValid = true;
-		this.props.elements.forEach((element) => {
-			if (formIsValid && element.isRequired && this.state[element.camelCaseName] !== 0
-				&& !this.state[element.camelCaseName]) {
+		let submitState = Object.assign({}, this.state);
+
+		formIsValid = this.validateElements(this.props.elements, this.state);
+
+		if (formIsValid && this.props.customVerification) {
+			const { camelCaseNames, verification } = this.props.customVerification;
+
+			if (Array.isArray(camelCaseNames)) {
+				const values = camelCaseNames.map(name => submitState[name]);
+				formIsValid = verification(values);
+			}
+			else { formIsValid = verification(submitState[camelCaseNames]) }
+		}
+
+		if (!formIsValid) return;
+
+		this.resetState();
+		this.props.onSubmit(submitState);
+		this.props.changeVisibility(false);
+	}
+
+	validateElements(elements, values) {
+		let formIsValid = elements.length > 0;
+
+		// contains all elements including nested elements that may be present in element.options
+		const iterableElements = elements.flatMap(element => (element.options && element.options.elements) || element);
+
+		iterableElements.forEach(element => {
+			const elementFromState = values[element.camelCaseName];
+
+			if (formIsValid && element.isRequired && elementFromState !== 0 && !elementFromState) {
 				Alert.alert(`Please input a value into the ${element.placeholder} field.`);
 				formIsValid = false;
 			}
+
+			if (formIsValid && element.isValid && element.isValid(elementFromState)) {
+				Alert.alert(`${element.placeholder} should be in the shape ${element.validForm}`);
+				formIsValid = false;
+			}
 		});
-		if (formIsValid) {
-			this.props.changeVisibility(false);
-			this.props.onSubmit(this.state);
-		}
+
+		return formIsValid;
+	}
+
+	resetState() {
+		let initialState = copyStateAndSetValuesToNull(this.state);
+		Object.assign(initialState, this.props.initialValues);
+		this.setState(initialState);
 	}
 
 	render() {
-		const { elements, title, changeVisibility, visible } = this.props;
+		const { elements, title, changeVisibility, visible, justElements } = this.props;
```

Comment:
what do you mean toggle-able? it takes away the header and buttons

Reply:
The header and button visibility is being toggled on or off. Maybe not actually but from the perspective of the user, it is. It could be `shouldRenderJustElements` or `shouldRenderElementsInPage` or idk